src/twister2/yaml_file.py

```diff
@@ -96,21 +96,20 @@ def _read_test_specifications_from_yaml(
         test_name: str  # type: ignore
         spec: dict  # type: ignore
 
-        for key, value in spec.items():
-            common_value = common.pop(key, None)
-            if not common_value:
+        for key, common_value in common.items():
+            if key not in spec:
```

Comment:
I think there is another bug here. If the first `if` statement is true we don't check the others, which should prepare value before assigning. This bug existed before your change. You can fix it in this PR or create an issue to fix it later. 
```python
if key not in spec:
    spec[key] = common_value
    continue
if key == 'filter':
    spec[key] = _join_filters([spec[key], common_value])
    continue
if isinstance(common_value, str):
    spec[key] = _join_strings([spec[key], common_value])
    continue
if isinstance(common_value, list):
    spec[key] = spec[key] + common_value
    continue
```

Reply:
Also it seems this is not covered by unit test.

Reply:
Is there? `if key not in spec` means that the given key was not mentioned in a scenario -> assign value from common to that field. All other checks only makes sense if the value was already present in a scenario. Since these ifs are exclusive, maybe changing followings to `elif` and omitting `continue`?

Reply:
It seems I was wrong.

Reply:
@fundakol What do you think about "Since these ifs are exclusive, maybe changing followings to elif and omitting continue would be more readable?" Do you agree?

Reply:
I rewrote it to more readable version (with `elif` statements) with additional comment.